### ISSUE TITLE
resize! is broken for LowStorageRK2NCache

### ISSUE THREAD

**Issue by ranocha** [SciML/OrdinaryDiffEq.jl#1277]
```julia
julia> using Test, OrdinaryDiffEq, DiffEqCallbacks

julia> function f(du, u, p, t)
           du .= u
           nothing
       end
f (generic function with 1 method)

julia> u0 = ones(2)

julia> condition(u, t, integrator) = t - 1
condition (generic function with 1 method)

julia> function affect!(integrator)
         u = integrator.u
         resize!(integrator, length(u) + 1)
         u .= 1
         nothing
       end
affect! (generic function with 1 method)

julia> callback = ContinuousCallback(condition, affect!);

julia> tspan = (0.0, 2.0)
(0.0, 2.0)

julia> prob = ODEProblem(f, u0, tspan)
ODEProblem with uType Array{Float64,2} and tType Float64. In-place: true
timespan: (0.0, 2.0)
u0: [1.0 1.0; 1.0 1.0]

julia> sol = solve(prob, RK4(), dt=0.125, adaptive=false, callback=callback)

julia> @test all(sol[end] .== sol[end][1])
Test Passed

julia> sol = solve(prob, CarpenterKennedy2N54(williamson_condition=false), dt=0.125, adaptive=false, callback=callback)

julia> @test all(sol[end] .== sol[end][1])
Test Failed at REPL[22]:1
  Expression: all(sol[end] .== (sol[end])[1])
ERROR: There was an error during testing
```

The `LowStorageRK2NCache` doesn't seem to be updated correctly. In our application in [Trixi.jl](https://github.com/trixi-framework/Trixi.jl), this results in stochastic errors when using adaptive mesh refinement. I will try to look into it but would be glad about any help.

**Comment by ranocha** [comment]
I propose to add something along the lines of 
```julia


# additional cache tests to find more bugs
println("Additional resize! checks")
u0resize3 = ones(2)
fresize3 = (du, u, p, t) -> du .= u
prob_resize3 = ODEProblem(fresize3, u0resize3, (0.0, 2.0))
condition_resize3 = (u, t, integrator) -> t - 1
affect!_resize3 = function (integrator)
  resize!(integrator, 3)
  integrator.u .= 1
  nothing
end
callback_resize3 = ContinuousCallback(condition_resize3, affect!_resize3)

for alg in CACHE_TEST_ALGS
  @show alg
  sol = solve(prob_resize3, alg, callback=callback_resize3, dt=0.125)
  @test size(sol[end]) == (3,)
  if alg isa Union{CarpenterKennedy2N54, HSLDDRK64}
    @test_broken all(sol[end] .== sol[end][1])
  else
    @test all(sol[end] .== sol[end][1])
  end
end
```
to `test/integrators/ode_cache_tests.jl`.

**Comment by ChrisRackauckas** [comment]
What's the current status on what you need?

**Comment by ranocha** [comment]
Currently, we need to have https://github.com/SciML/OrdinaryDiffEq.jl/pull/1278 merged.

**Comment by ranocha** [comment]
We might need a fix for #1279 in the future, but currently merging #1278 suffice for our needs.
